src/curvilinear.jl

```diff
@@ -455,19 +455,25 @@ function cornerindices(p::CurvilinearPolygon{T}) where {T}
     valid_ind = setdiff(1:length(p.p), curve_bound_ind)
     return valid_ind
 end
-function cornerindices(p::CurvilinearPolygon, p0::Vector{<:Point})
+function cornerindices(p::CurvilinearPolygon, p0::Vector{<:Point}; tol)
     isempty(p0) && return Int[]
     valid_ind = cornerindices(p)
     isempty(valid_ind) && return Int[]
-    # Pick the closest to p0
-    return valid_ind[map(p0) do px
-        idx = findfirst(p_idx -> isapprox(px, p_idx), p.p[valid_ind])
-        !isnothing(idx) && return idx
-        return findmin(norm.(p.p[valid_ind] .- px))[2]
-    end]
+    # Pick the closest to p0 satisfying tolerance.
```

Comment:
Is it possible to call `cornerindices(p::Vector{<:Point}, p0::Vector{<:Point}; tol)` here and then check against or map back to valid indices? I'm nervous about the selection code not being shared.